command/services/register/register.go

```diff
@@ -44,6 +45,8 @@ func (c *cmd) init() {
 		"Address of the service to register for arg-based registration.")
 	c.flags.IntVar(&c.flagPort, "port", 0,
 		"Port of the service to register for arg-based registration.")
+	c.flags.StringVar(&c.flagSocketPath, "socket", "",
```

Comment:
This new flag should also be documented in the website:
https://www.consul.io/commands/services/register#service-registration-flags